gibberish_server/controllers/room_controller.js

```diff
@@ -24,36 +26,48 @@ class RoomController {
 
     static joinRoom(req, res) {
         // TODO: race conditionSTATE.GAME_WAITING
-        let r = roomModel.Room.deserializeRoom(roomModel.getRoomInfoAsJson(roomId));
-        if (r.state != STATE.GAME_WAITING) return "";
```

Comment:
why return empty string?

Reply:
I have refactored this to send a 400 with a error message